broker/pipelines/cdn_dedicated_waf.py

```diff
@@ -79,6 +79,7 @@ def queue_all_cdn_dedicated_waf_update_tasks_for_operation(
         .then(letsencrypt.initiate_challenges, operation_id, **correlation)
         .then(route53.create_TXT_records, operation_id, **correlation)
         .then(route53.wait_for_changes, operation_id, **correlation)
+        .then(route53.remove_old_DNS_records, operation_id, **correlation)
```

Comment:
I did have some concern about where to put this task in the pipeline. Technically, after this step is complete, ALIAS records for the old domains from the service will no longer work. And the new ALIAS records for the updated domains aren't created until later in the pipeline.

So if all of the domains for a CDN service were changed, there would be downtime between when the old DNS records were removed and the new ones were created. I decided that this is okay, since if you're updating all the domains for an instance, you might expect some downtime.

Thoughts?

Reply:
changing _all_ the domains on an instance seems weird to me. 
I *think* the order doesn't matter too much, but maybe it makes the most sense either just before or just after removing the old cert (since that's where a request to the old domain is going to start breaking no matter what)

Reply:
In this pipeline, `update_distribution` is what changes the value of `current_certificate`. So you think this task should happen after that?

Reply:
seems like right around there would be the most perfect time, but I also don't think it matters a whole lot. I think the expectation should be once you call update-service removing that domain, the domain will no longer reliably work